spec/requests/profile_spec.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+describe "Profile", type: :request do
+  describe "#index" do
+    context "when user is not logged in" do
+      before { get profile_index_path, params: {uid: "user-id"} }
+      it { expect(response).to redirect_to(new_user_session_path) }
+    end
+
+    context "when user is logged in" do
+      let(:user1) { create(:user1) }
+      before { sign_in user1 }
+      it "renders index template" do
+        get profile_index_path, params: {uid: user1.uid}
+        expect(response).to render_template(:index)
+      end
+
+      context "when profile belongs to logged in user" do
+        it "assigns profile instance to be that of logged in user" do
+          get profile_index_path, params: {uid: user1.uid}
+          expect(assigns(:profile)).to eq(user1)
```

Comment:
```suggestion
          expect(response.body).to include(user.name)
```

Since `@profile` references the user and profile's `index.html.erb` contains a title of the user's name, we can check that the response body contains the user's name.

Reply:
Good call